### PR TITLE
Migrate repo's PROJECT to Kubebuilder v2

### PR DESCRIPTION
**Pull request by nader-ziada**
**What this PR does / why we need it**:
- migrate the PROJECT file to the `kubebuilder` v2 structure
- the project directory structure and markers already follows the v2 format
- following kubebuilder [migration guide](https://book.kubebuilder.io/migration/guide.html) 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -1,3 +1,22 @@
-version: "1"
-domain: k8s.io
+version: "2"
+domain: x-k8s.io
```
Comment (CecileRobertMichon) [inline PROJECT]: why do we need to change the domain?

Comment (CecileRobertMichon) [reply PROJECT]: (don't think it's wrong, just curious)

Comment (nader-ziada) [reply PROJECT]: The crds (https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/config/crd/bases/infrastructure.cluster.x-k8s.io_azureclusters.yaml) are already using the `x-k8s`

**Comment by nader-ziada** [comment]
did a rebase to resolve conflicts

**Pull request merged**